CONTRIBUTING.rst

```diff
@@ -0,0 +1,313 @@
+What should I know before I get started?
+========================================
+
+The structure of pyiron
+-----------------------
+
+The principles of pyiron
+------------------------
+
+How can I contribute?
+=====================
+
+Reporting bugs
+--------------
+
+    Note: If you find a closed issue that seems like it is the same
+    thing that you're experiencing, open a new issue and include a
+    link to the original issue in the body of your new one.
+
+**Before Submitting A Bug Report**
+
+Check if you can reproduce the problem in the latest version of pyiron.
+Check the `FAQ page`_ for a list of common questions and problems.
+Briefly search the issues page for `bugs`_  to see if the problem has already
+been reported. If it has and the issue is still open, add a comment
+to the existing issue instead of opening a new one.
+
+**How Do I Submit A (Good) Bug Report?**
+
+Bugs are tracked as GitHub issues. You can create an issue on
+the pyiron repository by including the following information:
+
+* Use a clear and descriptive title for the issue to identify the problem.
+* Describe the exact steps you took so we can reproduce the problem as closely as possible.
+* Provide sample code that causes the problem. Include code snippets as markdown code blocks.
+* Include information about the environment (OS, python version, how packages were installed) in which you were running pyiron.
+* Explain what you expected to happen, and what happened instead.
+
+Suggesting Enhancements
+-----------------------
+
+**How Do I Submit A (Good) Enhancement Suggestion?**
+
+Enhancement suggestions are tracked as GitHub issues. You can create an issue on
+the pyiron repository by including the following information:
+
+* Use a clear and descriptive title for the issue to identify the suggestion.
+* Describe the exact behavior you would expect the suggested feature to produce.
+* Provide sample code that you would use to access the feature. If possible, include code for how you think the feature could be built into pyiron's codebase. Include code snippets as markdown code blocks.
+
+Your first code contribution
+----------------------------
+
+Unsure where to begin contributing to pyiron? You can start by looking
+through these good-first-issue and help-wanted issues:
+
+* `Good first issues`_ - issues which should only require a few lines of code, and a test or two.
+* `Help wanted issues`_ - issues which should be a bit more involved than beginner issues.
+
+**Local development**
+
+pyiron can be developed and tested locally. If you are using pyiron to run an
+external software package, e.g. VASP or LAMMPS, you might also need to install
+those packages locally to run certain integration tests in pyiron.
+
+To get the developmental (git) version of pyiron,
+
+```
+git clone https://github.com/pyiron/pyiron.git
+```
+If you don't already have the dependencies installed in your environment,
+```
+cd pyiron
+pip install .
```

Comment:
Wouldn't it be more consistent to install all the packages through `conda-forge` first and then overwrite the python path to use the your local branch of pyiron?

Reply:
Yeah doing it with conda does seem more elegant, but I admit I am sometimes confused by the interaction between conda and $PYTHONPATH.

Maybe conda develop is the way to go? e.g.
```bash
conda create --name pyiron_dev
conda activate pyiron_dev
conda install -c conda-forge conda-build pyiron
conda remove --force pyiron  # leaves dependencies untouched
conda develop /path/to/git/installed/pyiron  # adds git folder to site-packages/conda.pth file
```